cc/crypto/client_encryptor.cc

```diff
@@ -17,7 +17,73 @@
 #include "cc/crypto/client_encryptor.h"
 
 #include "absl/status/statusor.h"
+#include "cc/crypto/common.h"
 #include "cc/crypto/hpke/sender_context.h"
 #include "oak_crypto/proto/v1/crypto.pb.h"
 
-namespace oak::crypto {}  // namespace oak::crypto
+namespace oak::crypto {
+
+namespace {
+using oak::crypto::v1::EncryptedRequest;
+using oak::crypto::v1::EncryptedResponse;
+}  // namespace
+
+absl::StatusOr<std::unique_ptr<ClientEncryptor>> ClientEncryptor::Create(
+    absl::string_view serialized_server_public_key) {
+  absl::StatusOr<SenderContext> setup_base_sender_result =
+      SetupBaseSender(serialized_server_public_key, OAK_HPKE_INFO);
+  if (!setup_base_sender_result.ok()) {
+    return setup_base_sender_result.status();
+  }
+  SenderContext sender_context = std::move(*setup_base_sender_result);
+  return std::make_unique<ClientEncryptor>(ClientEncryptor(sender_context));
```

Comment:
keep using setup_base_sender_result variable. The absl::StatusOr class is similar to pointers. If you want to access the object (assuming status is ok()), you can get the object using *

Reply:
Done